data/campaigns/Heir_To_The_Throne/scenarios/13_The_Dwarven_Doors.cfg

```diff
@@ -160,7 +160,70 @@
         #
         {VARIABLE_OP true_entrance_location rand "1..2"}
 
-        # TODO: manually add snow detritus
+        [lua]
+            code=<<
+
+local helper = wesnoth.require "lua/helper.lua"
+
+-- The only parameter right now is the coverage (real 0.0 to 1.0, incl.)
+-- For prime-time use the filter terrain needs to come in and the convert
+-- table needs to be VERY complete. Or, rename this to morph_terrain and
+-- have the conversion table come in as an argument, too.
+
+local filter = {}
+filter.terrain = 'Ce,Ch,Chr,Co,Gd,Gd^Vc,Gd^Vh,Gd^Vo,Gll^Fp,Hhd,Hhd^Fp,Hhd^Vo,Ke,Khr,Ko,Md,Mm,Rd,Re,Re^Vo,Rr,Ww'
+local locations = wesnoth.get_locations(helper.shallow_parsed(filter))
```

Comment:
So, instead of these three lines, I think I'd prefer:

```
local locations = wesnoth.get_locations{terrain = '...'}
```

This:
- Builds the table all at once, instead of creating it and then adding a key (doesn't really make a difference, especially with just one key, but it's my stylistic preference).
- Avoids the unnecessary call to shallow_parsed. That call won't do anything anyway (since you're passing a plain table, not a userdata vconfig), but I think it's silly to call it when you know it's not needed. It's intended for when you're not sure whether you have a vconfig or a plain table.
- Uses one less local variable. This isn't really important though. (There's really no reason to use fewer locals.)